lib/libutils/ext/trace.c

```diff
@@ -91,30 +117,26 @@ void trace_printf(const char *function, int line, int level, bool level_ok,
 		return;
 	boffs += res;
 
-	/* Print the Thread ID */
-	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
-		thread_id = -1;
-	else
-		thread_id = trace_ext_get_thread_id();
-
-	res = print_thread_id(buf + boffs, sizeof(buf) - boffs, thread_id);
-
-	if (res < 0)
-		return;
-	boffs += res;
+	if (level_ok && (BIT(level) & CFG_MSG_LONG_PREFIX_MASK)) {
+		/* Print the core ID if in atomic context  */
+		res = print_core_id(buf + boffs, sizeof(buf) - boffs);
+		if (res < 0)
+			return;
+		boffs += res;
 
-	/* Print the function and line */
-	if (level_ok && !(BIT(level) & CFG_MSG_LONG_PREFIX_MASK))
-		function = NULL;
+		/* Print the Thread ID */
+		thread_id = trace_ext_get_thread_id();
+		res = print_thread_id(buf + boffs, sizeof(buf) - boffs, thread_id);
+		if (res < 0)
+			return;
+		boffs += res;
 
-	if (function) {
```

Comment:
You can't remove this test because it does happen that `function == NULL` (see `*MSG_RAW`) and in this case we don't want to print `(null):0`.